azure/scope/managedcontrolplane_test.go

```diff
@@ -1619,3 +1621,156 @@ func TestManagedControlPlaneScope_AutoUpgradeProfile(t *testing.T) {
+					AdditionalTags: make(infrav1.Tags, 0),
+				},
+				&groups.GroupSpec{
+					Name:           "different-rg",
+					ClusterName:    "cluster1",
+					Location:       "my-location",
+					AdditionalTags: make(infrav1.Tags, 0),
+				},
+			},
+		},
+		{
+			name: "virtualNetwork belongs to a same resource group",
+			input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						ResourceGroupName: "dummy-rg",
+						AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+							Location: "my-location",
+							VirtualNetwork: infrav1.ManagedControlPlaneVirtualNetwork{
+								ResourceGroup: "dummy-rg",
+								ManagedControlPlaneVirtualNetworkClassSpec: infrav1.ManagedControlPlaneVirtualNetworkClassSpec{
+									Name:      "vnet1",
+									CIDRBlock: "10.1.0.0/8",
+									Subnet: infrav1.ManagedControlPlaneSubnet{
+										Name:      "subnet1",
+										CIDRBlock: "10.1.0.0/16",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: []azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup]{
+				&groups.GroupSpec{
+					Name:           "dummy-rg",
+					ClusterName:    "cluster1",
+					Location:       "my-location",
+					AdditionalTags: make(infrav1.Tags, 0),
+				},
+			},
+		},
+		{
+			name: "virtualNetwork resource group not specified",
+			input: ManagedControlPlaneScopeParams{
+				Cluster: &clusterv1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+				},
+				ControlPlane: &infrav1.AzureManagedControlPlane{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "cluster1",
+						Namespace: "default",
+					},
+					Spec: infrav1.AzureManagedControlPlaneSpec{
+						ResourceGroupName: "dummy-rg",
+						AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+							Location: "my-location",
+							VirtualNetwork: infrav1.ManagedControlPlaneVirtualNetwork{
+								ManagedControlPlaneVirtualNetworkClassSpec: infrav1.ManagedControlPlaneVirtualNetworkClassSpec{
+									Name:      "vnet1",
+									CIDRBlock: "10.1.0.0/8",
+									Subnet: infrav1.ManagedControlPlaneSubnet{
+										Name:      "subnet1",
+										CIDRBlock: "10.1.0.0/16",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: []azure.ASOResourceSpecGetter[*asoresourcesv1.ResourceGroup]{
+				&groups.GroupSpec{
+					Name:           "dummy-rg",
+					ClusterName:    "cluster1",
+					Location:       "my-location",
+					AdditionalTags: make(infrav1.Tags, 0),
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		c := c
```

Comment:
Out of curiosity, does the linter complain here without this line?

Reply:
Linter does not complain.